### PR TITLE
Fix infinite loop in `mj_deleteFileVFS`

### PR DESCRIPTION
**Pull request by hartikainen**
In its current form, the `mj_deleteFileVFS` results in infinite loop with more than one file. This should fix the issue.

### CODE REVIEW

File: src/engine/engine_vfs.c [hunk 1]
```diff
@@ -179,10 +179,10 @@ int mj_deleteFileVFS(mjVFS* vfs, const char* filename) {
       mju_free(vfs->filedata[i]);
 
       // scroll remaining files forward
-      while (i<vfs->nfile-1) {
-        mjSTRNCPY(vfs->filename[i], vfs->filename[i+1]);
-        vfs->filesize[i] = vfs->filesize[i+1];
-        vfs->filedata[i] = vfs->filedata[i+1];
+      for (int j = i; j < vfs->nfile - 1; ++j) {
```
Comment (yuvaltassa) [inline src/engine/engine_vfs.c:182]: C files in `engine/` have a somewhat [idiosyncratic coding convention](https://github.com/deepmind/mujoco/blob/main/STYLEGUIDE.md). I'm not going to say that I love it, but it is what it is and consistency trumps everything when style is concerned. To wit, `for (int j=i; j<vfs->nfile-1; j++)` would be consistent with the rest of the C code.

This comment does not apply to other code (e.g. test files).

Comment (saran-t) [inline src/engine/engine_vfs.c:182]: This loop should start at i+1 not i.

Comment (hartikainen) [reply src/engine/engine_vfs.c]: I think `i` is correct. We delete the `i`th file and then always move e.g. `vfs->filesize[j] = vfs->filesize[j+1];`.

Comment (hartikainen) [reply src/engine/engine_vfs.c]: Fixed.

**Review by hartikainen — COMMENTED** [review]
Changed

File: src/engine/engine_vfs.c (hunk 1 above)
Comment (saran-t) [reply src/engine/engine_vfs.c]: You're right.

**Review by saran-t — APPROVED** [review]

**Pull request merged**